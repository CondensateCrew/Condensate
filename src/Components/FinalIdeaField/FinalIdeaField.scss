@import '../../css/index.scss';
@import '../../css/variables.scss';

.final-idea-field-section {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 85%;
}

.final-idea-question-div {
  display: flex;
  justify-content: space-evenly;
  width: 62%;
  border-radius: $one-line-border-radius;
  box-shadow: $external-shadow;
  padding: $medium-space;
  margin: $small-space;
  img {
    width: 1.25rem;
  }
}

.final-idea-field-div {
  width: 62%;
  height: 40vh;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: column;
  border-radius: $one-line-border-radius;
  box-shadow: $external-shadow;
  padding: $medium-space;

  textarea {
    resize: none;
    width: 95%;
    height: 25vh;
    padding: $medium-space;
    box-shadow: $inner-shadow;
    border-radius: $one-line-border-radius;
  }

  button {
    @extend %regular-btn;
  }
}