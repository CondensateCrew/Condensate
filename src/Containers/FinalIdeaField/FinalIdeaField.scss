@import '../../css/index.scss';
@import '../../css/variables.scss';

.final-idea-field-section {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 85%;
}

.final-idea-question-div {
  display: flex;
  justify-content: space-evenly;
  width: 40vw;
  background: $primary-gradient;
  border-radius: $one-line-border-radius;
  box-shadow: $external-shadow;
  padding: $medium-space;
  margin: $small-space;
  img {
    width: $big-text;
  }
}

.final-idea-field-div {
  background: $primary-gradient;
  width: 40vw;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: column;
  border-radius: $one-line-border-radius;
  box-shadow: $external-shadow;
  padding: $big-block-padding-space + $tiny-space;
  textarea {
    resize: none;
    width: 100%;
    height: 25vh;
    padding: $medium-space;
    box-shadow: $inner-shadow;
    border-radius: $one-line-border-radius;
    margin-bottom: $small-space;
  }
  button {
    @extend %regular-btn;
  }
}
