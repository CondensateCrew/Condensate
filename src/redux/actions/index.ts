import {
  Action,
  Category,
  Brainstorm,
  UserLoginReceived,
  RandomWordCollection,
  Insight,
  ActionObject,
  IAddUserAction,
  ILogoutUserAction,
  IAddAllActionsAction,
  IAddNewActionAction,
  IAddAllBrainstormsAction,
  IAddNewBrainstormAction,
  IDeleteBrainstormAction,
  IAddAllCategoriesAction,
  IAddNewCategoryAction,
  IAddRandomWordCollectionsAction,
  IAddQuestionTemplatesAction,
  IAddSecretSauceAction,
  IAddCurrentBrainstormAction,
  IRemoveCurrentBrainstormAction,
  IAddChosenWordAction,
  IRemoveChosenWordAction,
  IAddInsightAction,
  IAddQueryAction,
  IRemoveQueryAction,
  IAddFilterAction,
  IRemoveFilterAction,
  IEndTimeAction,
  IReverseTimeAction,
  IEndInstructionsAction,
  IReverseInstructionsAction
} from 'interfaces';

// user reducer action creators
export const addUser = (user: UserLoginReceived): IAddUserAction => ({
  type: 'ADD_USER',
  user: user
});

export const logOutUser = (): ILogoutUserAction => ({
  type: 'LOG_OUT_USER',
});

// actions reducer action creators
export const addAllActions = (actions: Action[]): IAddAllActionsAction => ({
  type: 'ADD_ALL_ACTIONS',
  actions: actions
});

export const addNewAction = (action: Action): IAddNewActionAction => ({
  type: 'ADD_NEW_ACTION',
  action: action
});

// allBrainstorms reducer action creators
export const addAllBrainstorms = (brainstorms: Brainstorm[]): IAddAllBrainstormsAction => ({
  type: 'ADD_ALL_BRAINSTORMS',
  brainstorms: brainstorms
});

export const addNewBrainstorm = (brainstorm: Brainstorm): IAddNewBrainstormAction => ({
  type: 'ADD_BRAINSTORM',
  brainstorm: brainstorm
});

export const deleteBrainstorm = (id: number): IDeleteBrainstormAction => ({
  type: 'DELETE_BRAINSTORM',
  id: id
});

// categories reducer action creators
export const addAllCategories = (categories: Category[]): IAddAllCategoriesAction => ({
  type: 'ADD_ALL_CATEGORIES',
  categories: categories
});

export const addNewCategory = (category: Category): IAddNewCategoryAction => ({
  type: 'ADD_CATEGORY',
  category: category
});

// randomWordCollections reducer action creators
export const addRandomWordCollections = (collections: RandomWordCollection[]): IAddRandomWordCollectionsAction => ({
  type: 'ADD_ALL_COLLECTIONS',
  collections: collections
});

// questionTemplates reducer action creators
export const addQuestionTemplates = (templates: string[]): IAddQuestionTemplatesAction => ({
  type: 'ADD_ALL_TEMPLATES',
  templates: templates
});

// secretSauce reducer action creators
export const addSecretSauce = (secretSauce: string[]): IAddSecretSauceAction => ({
  type: 'ADD_SECRETE_SAUSE',
  secretSauce: secretSauce
});

// currentBrainstorm reducer action creators
export const addCurrentBrainstorm = (currentBrainstorm: Brainstorm): IAddCurrentBrainstormAction => ({
  type: 'ADD_CURRENT_BRAINSTORM',
  currentBrainstorm: currentBrainstorm
});

export const removeCurrentBrainstorm = (): IRemoveCurrentBrainstormAction => ({
  type: 'REMOVE_CURRENT_BRAINSTORM'
});

// chosenWords reducer action creators
export const addChosenWord = (chosenWord: string): IAddChosenWordAction => ({
  type: 'ADD_WORD',
  chosenWord: chosenWord
});

export const removeChosenWord = (chosenWord: string): IRemoveChosenWordAction => ({
  type: 'REMOVE_WORD',
  chosenWord: chosenWord
});

// insights reducer action creators
export const addInsight = (insight: Insight): IAddInsightAction => ({
  type: 'ADD_INSIGHT',
  insight: insight
});

// query reducer action creators
export const addQuery = (query: string): IAddQueryAction => ({
  type: 'ADD_QUERY',
  query: query
});

export const removeQuery = (): IRemoveQueryAction => ({
  type: 'REMOVE_QUERY'
});

// filter reducer action creators
export const addFilter = (filter: string): IAddFilterAction => ({
  type: 'ADD_FILTER',
  filter: filter
});

export const removeFilter = (): IRemoveFilterAction => ({
  type: 'REMOVE_FILTER'
});

// timeEnded reducer action creators
export const endTime = (): IEndTimeAction => ({
  type: 'END_TIME'
});

export const reverseTime = (): IReverseTimeAction => ({
  type: 'REVERSE_TIME'
});

// instructionsEnded reducer action creators
export const endInstructions = (): IEndInstructionsAction => ({
  type: 'END_INSTRUCTIONS'
});

export const reverseInstructions = (): IReverseInstructionsAction => ({
  type: 'REVERSE_INSTRUCTIONS'
});
