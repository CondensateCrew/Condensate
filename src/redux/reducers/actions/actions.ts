import { ActionObject } from 'interfaces';

type state = string[] | [ ];

const actions = (state: state = [ ], action: ActionObject) => {
  switch(action.type) {
    case 'ADD_ALL_ACTIONS':
      return action.actions;
    case 'ADD_NEW_ACTION':
      return [...state, action.action];
    case 'CLEAN_STORE':
      return [ ];
    default:
      return state;
  }
}

export default actions;
