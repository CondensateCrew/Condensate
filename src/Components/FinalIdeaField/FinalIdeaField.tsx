import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { postBrainstorm } from 'apiCalls/apiCalls';
import { AppStore } from 'interfaces';
import './FinalIdeaField.scss';
// import edit from '../../assets/edit.svg';
import { useHistory } from 'react-router-dom';

const FinalIdeaField:React.FC = () => {
  const [ brainstormIdea, setBrainstormIdea ] = useState<string>('');
  const { id, action, isGenius, question, categories } = useSelector((state: AppStore) => ({
    id: state.user.id,
    action: state.currentBrainstorm.action,
    isGenius: state.currentBrainstorm.isGenius,
    question: state.currentBrainstorm.question,
    categories: state.currentBrainstorm.categories
  }));
  let history = useHistory();

const originalQuestion = useSelector((state:AppStore) => state.currentBrainstorm.question)

  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setBrainstormIdea(e.target.value)
  }

  const postBrainstormIdea = async () => {
    const brainstorm = {
      idea: brainstormIdea,
      id,
      action,
      isGenius,
      question,
      categories
    }

    const options = {
      method: 'POST',
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(brainstorm)
    }

    const res = await postBrainstorm(options) //eslint-disable-line
    history.push('/dashboard')
  }

  return (
    <section className='final-idea-field-section'>
      <div className='final-idea-question-div'>
        <h3>{originalQuestion}</h3>
        {
          // <img src={edit} alt='pencil'/>
        }
      </div>
      <div className='final-idea-field-div'>
        <textarea onChange={handleChange} value={brainstormIdea}
        placeholder="Type your idea here..."></textarea>
        <button onClick={postBrainstormIdea}>Post Idea</button>
      </div>
    </section>
  )
}

export default FinalIdeaField;
