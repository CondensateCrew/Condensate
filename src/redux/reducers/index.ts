import { combineReducers } from 'redux';
import user from '../reducers/user/user';
import allBrainstorms from '../reducers/allBrainstorms/allBrainstorms';
import categories from '../reducers/categories/categories';
import actions from '../reducers/actions/actions';
import randomWordCollections from '../reducers/randomWordCollections/randomWordCollections';
import questionTemplates from '../reducers/questionTemplates/questionTemplates';
import secretSauce from '../reducers/secretSauce/secretSauce';
import currentBrainstorm from '../reducers/currentBrainstorm/currentBrainstorm';
import chosenWords from '../reducers/chosenWords/chosenWords';
import insights from '../reducers/insights/insights';
import query from '../reducers/query/query';
import filter from '../reducers/filter/filter';
import timeEnded from '../reducers/timeEnded/timeEnded';
import instructionsEnded from '../reducers/instructionsEnded/instructionsEnded';

const rootReducer = combineReducers({
  user,
  allBrainstorms,
  categories,
  actions,
  randomWordCollections,
  questionTemplates,
  secretSauce,
  currentBrainstorm,
  chosenWords,
  insights,
  query,
  filter,
  timeEnded,
  instructionsEnded
});

export default rootReducer;
