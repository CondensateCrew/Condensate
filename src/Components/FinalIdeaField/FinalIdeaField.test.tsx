/**
 * @jest-environment jsdom
 */

import React from 'react';
import FinalIdeaField from './FinalIdeaField';
import { mount } from 'enzyme';
import { addCurrentBrainstorm } from 'redux/actions';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => 'question?'
}));
jest.mock('redux/actions');

describe('FinalIdeaField component', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(<FinalIdeaField />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addCurrentBrainstorm after rendering', () => {
    expect(addCurrentBrainstorm).toHaveBeenCalled();
  });
});
