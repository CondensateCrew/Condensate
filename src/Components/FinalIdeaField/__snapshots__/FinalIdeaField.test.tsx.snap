// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalIdeaField component should match the snapshot 1`] = `
<FinalIdeaField>
  <section
    className="final-idea-field-section"
  >
    <div
      className="final-idea-question-div"
    >
      <h3>
        question?
      </h3>
    </div>
    <div
      className="final-idea-field-div"
    >
      <textarea
        onChange={[Function]}
        placeholder="Type your idea here..."
        value=""
      />
      <button
        onClick={[Function]}
        type="button"
      >
        Post Idea
      </button>
    </div>
  </section>
</FinalIdeaField>
`;
